src/components/ProposalStatus/TimeDetail.tsx

```diff
@@ -0,0 +1,48 @@
+import { ProposalState } from 'types/types.guilds.d';
+import { TimeDetailProps } from './types';
+import moment, { unix } from 'moment';
+import { useGuildConfig } from 'Modules/Guilds/Hooks/useGuildConfig';
+
+export const TimeDetail: React.FC<TimeDetailProps> = ({
+  endTime,
+  timeDetail,
+  status,
+  guildId,
+}) => {
+  const { data: guildConfig } = useGuildConfig(guildId);
+  const ONE_MINUTE = 60000;
+
+  if (!guildConfig || status === ProposalState.Active) {
+    return (
+      <span title={endTime?.format('MMMM Do, YYYY - h:mm a')}>
+        {timeDetail}
+      </span>
+    );
+  }
+
+  let { timeForExecution }: any = guildConfig;
+  let executionTimeDetail = '';
+
+  timeForExecution = unix(timeForExecution.toNumber());
+  const executionTime = moment(endTime).add(timeForExecution);

```

Comment:
❓ There are a lot of variables changing here and it becomes a bit hard to read. Would it be possible to move the logic to functions in a separate file instead and try to keep the component clean from most of the logic?

Also would it be feasible to add some unit tests to this?